synapse/models/syn.py

```diff
@@ -23,7 +23,11 @@ def _normPyStr(self, text):
             if user is not None:
                 return user.iden, {}
 
-        return s_types.Guid._normPyStr(self, text)
+        try:
+            return s_types.Guid._normPyStr(self, text)
+        except s_exc.BadTypeValu:
+            mesg = f'No user named {text}.'
+            raise s_exc.NoSuchUser(mesg=mesg, username=text) from None
```

Comment:
Changing these exceptions to `BadTypeValu` should allow try syntax like this to continue to work `?=`

```suggestion
            raise s_exc.BadTypeValu(mesg=mesg, name=self.name, valu=text) from None
```